app/client-v2/src/views/HomeView.vue

```diff
@@ -0,0 +1,160 @@
+<script setup lang="ts">
+import { getApiUrl } from '@/config';
+import { useFetch, useDateFormat } from '@vueuse/core';
+import { ref } from 'vue';
+import InputGroup from 'primevue/inputgroup';
+import { FilterMatchMode } from 'primevue/api';
+import type { DataTableRowEditSaveEvent } from 'primevue/datatable';
+import type { ProjectsResponse } from '@/types/projectTypes';
+import { useToast } from 'primevue/usetoast';
+import Toast from 'primevue/toast';
+
+const toast = useToast();
+const showSuccessToast = (msg: string) => {
+  toast.add({ severity: 'success', summary: 'Success', detail: msg, life: 3000 });
+};
+const showErrorToast = (msg: string) => {
+  toast.add({ severity: 'error', summary: 'Error', detail: msg, life: 3000 });
+};
+const apiUrl = getApiUrl();
+const {
+  data,
+  error: projectsError,
+  isFetching,
+  execute: refetchProjects
+} = useFetch(apiUrl + '/projects', { credentials: 'include' }).json<ProjectsResponse>();
+
+const filters = ref({
+  name: { value: null, matchMode: FilterMatchMode.CONTAINS }
+});
+
+const newProjectName = ref('');
+const addProject = async () => {
+  if (!newProjectName.value) return;
+  const { error } = await useFetch(apiUrl + '/project', {
+    credentials: 'include'
+  })
+    .post({ name: newProjectName.value })
+    .json();
+
+  if (error.value) {
+    showErrorToast('Error creating project');
+    return;
+  }
+  newProjectName.value = '';
+  showSuccessToast('Project created successfully');
+  refetchProjects();
+};
+
+const editingRows = ref([]);
+const onRowEditSave = async (event: DataTableRowEditSaveEvent) => {
+  let { newData, index } = event;
+
+  if (data.value !== null && data.value.data[index].name !== newData.name) {
+    const { error } = await useFetch(apiUrl + `/project/${data?.value?.data[index].id}/rename`, {
+      credentials: 'include'
+    })
+      .post({ name: newData.name })
+      .json();
+    if (error.value) {
+      showErrorToast('Error renaming project');
+      return;
+    }
+    data.value.data[index] = newData;
+    showSuccessToast('Project renamed successfully');
+    refetchProjects();
```

Comment:
Do you need to both update the appropriate row, and also refetch projects? Could just have one or the other couldn't you?

Reply:
thats true have updaetd to just refetch